### PR TITLE
Update requirements for Ubuntu 18.04 & apache 2.4

### PR DESCRIPTION
**Pull request by andrerom**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | 
| Versions      | 2.2

1. Update requirements for v2.2 to reflect the new Ubuntu 18.04LTS release
2. Clarify that we now recommend event MPM for use with Apache, which implies usage of php-fpm for fastcgi setup
3. Some other minor fixes for v2

Our provided apache config has been updated to reflect this change: https://github.com/ezsystems/ezplatform/pull/300

#### DOC / QA
Point 2 has an impact on you both, install doc/testing for linux should be updated to reflect usage of Apache event MPM, instead of `prefork` + `mod_php` which is done here: https://github.com/ezsystems/developer-documentation/blob/cdeec99192d66af91d6f105cf90f704236b152f1/docs/getting_started/install_manually.md#8-server-configuration

TL;DR; 1. prefork uses more memory as it loads php for all requests, even static once _(css, js, ..)_, 2. each child process is only able to handle one request at a time. Event MPM solves both issues.

###### A bit more background
For many years prefork was the way to configure PHP, where php lived as a module in apache. Back in 2010 php-fpm came, a official way to run PHP via fastcgi which was the only way to run PHP in nginx and lighttpd, Apache 2.4 actually has become the default in all our supported linux distroes over the last few years, and with it event MPM was made stable and the preferred way to run PHP in production. So we are a bit behind the curve on this one, simply because we have mainly been using nginx for the last few years ourselves.

### CODE REVIEW

File: docs/getting_started/requirements_and_system_configuration.md [hunk 1]
```diff
@@ -37,18 +37,21 @@ Supported setups are those we perform automated testing on. For security and per
...
     -   MySQL 5.5.3 and higher, 5.6\*, 5.7\*
     -   MariaDB 5.5, 10.0, 10.1, 10.2\*
 -   PHP
     -   7.1
+    -   7.2
 
 - Cluster
-    - Redis or Memcached
-    - Solr or SQL *(but does not provide same featureset or performance as Solr)*
+    - Redis or Memcached *(Not recommended as it has large performance issues with cache tagging by Symfony Cache component)*
```
Comment (SylvainGuittard) [inline docs/getting_started/requirements_and_system_configuration.md]: If I am right only `Memcached` has performances issues, right?
Maybe we should remove it.

Comment (andrerom) [reply docs/getting_started/requirements_and_system_configuration.md]: We can, we can rather mention it among experimental setups until the performance issue is solved in Symfony.

Comment (andrerom) [reply docs/getting_started/requirements_and_system_configuration.md]: fixed in 5f3a10f

Comment (andrerom) [reply docs/getting_started/requirements_and_system_configuration.md]: And some more in 4afed93

Comment (SylvainGuittard) [reply docs/getting_started/requirements_and_system_configuration.md]: Thanks. Definitely better.

**Comment by andrerom** [comment]
Regarding Apache 2.4 config, here are some pointers from Slack conversation with @dspe:
```
    <Directory /var/www/ez/web>
        <FilesMatch \.php$>
            # For serving php files configure mod_proxy to talk to php-fpm
            # See: https://wiki.apache.org/httpd/PHP-FPM#apache_httpd_2.4

            #! Socket use requires Apache 2.4.9+, Linux, & Apache + php-fpm is on the same machine
            SetHandler "proxy:unix:/var/run/php5-fpm.sock|fcgi://localhost/"
            #! If you need to use TCP for some reason that would instead be something like:
            # SetHandler "proxy:fcgi://localhost/:9000"
        </FilesMatch>

        Options FollowSymLinks
        AllowOverride None
        Require all granted
    </Directory>
```

And https://wiki.apache.org/httpd/PHP-FPM#Proxy_via_handler which suggested configuring proxy like this, probably implying `max=10` should be same number as configured php-fpm processes:
`<Proxy "fcgi://localhost:9000/" enablereuse=on max=10>`

**Comment by andrerom** [comment]
@DominikaK The Apache config has been updated in https://github.com/ezsystems/ezplatform/pull/300 to reflect this change, so you can remove the Special case label as this is now unblocked.

**Comment by andrerom** [comment]
@DominikaK This can go into 1.13 as we talked about, but when doing that do this:
- Changes in `Development and Experimental setups` => skip these changes, they are not relevant for 1.13
- Remove occurrences of `(prefork mode)`
- Remove occurrences of `(for use with nginx)`
- Remove occurrences of `(via libapache2-mod-php5 for Apache)`


Update Server list to (updates recommendation to allow Ubuntu 18.04, and drops recommendation for PHP5.6 installs as they don't perform well enough for 1.x UI):


||Debian|Ubuntu|RHEL / CentOS|
|------|------|------|------|
|Operating system|9.x "Stretch"|16.04LTS "Xenial" /  18.04LTS "Bionic"|7.x|
|Web Server|Nginx 1.10</br>Apache 2.4|Nginx 1.10 / 1.14</br>Apache 2.4|Nginx 1.10 *(latest via [RHSCL](https://access.redhat.com/documentation/en/red-hat-software-collections/))*</br>Apache 2.4|
|DBMS|MariaDB 10.0 / 10.1</br>MySQL 5.5|MariaDB 10.0 / 10.1</br>MySQL 5.7\*|MariaDB 10.1 *(latest via RHSCL)*</br>MariaDB 10.0 *(latest via RHSCL)*</br>MySQL 5.6 *(latest via RHSCL)*</br>MariaDB 5.5|
|PHP|PHP 7.0|PHP 7.0 / 7.2|PHP 7.0 *(latest via RHSCL)|
|PHP packages|php-cli</br>php-fpm</br>php-mysql or php-pgsql</br>php-xml</br>php-intl</br>php-curl</br>php-gd *or* php-imagick|php-cli</br>php-fpm *(for use with nginx)*</br>php-mysql or php-pgsql</br>php-xml</br>php-mbstring</br>php-intl</br>php-mcrypt</br>php-curl</br>php-gd or php-imagick|php-cli</br>php-fpm</br>php-mysqlnd or php-pgsql</br>php-xml</br>php-mbstring</br>php-process</br>php-intl</br>php-pear *(optional, provides pecl)*</br>php-gd or php-imagick *(via [pecl](https://pecl.php.net/package/imagick))*|
|Cluster PHP packages</br>*Also recommended for single server setup, improves performance of cache clearing operations.*|php-memcached *(via [pecl](https://pecl.php.net/package/memcached)) or* php-redis *(via [pecl](https://pecl.php.net/package/redis))*|php-memcached *(via [pecl](https://pecl.php.net/package/memcached)) or* php-redis *(via [pecl](https://pecl.php.net/package/redis))*|php-memcached *(via [pecl](https://pecl.php.net/package/memcached)) or* php-redis *(via [pecl](https://pecl.php.net/package/redis))*|

**Comment by dspe** [comment]
Hi guys, just a quick question : whats about the support of MariaDB 10.2 and Mysql > 5.7 ?
